File: OOP_final_project_python_version/pessoa.py
class Pessoa:
    def __init__(self, nome_oficial, cpf_cnpj, endereco, telefone):
        self._nome_oficial = nome_oficial
        self.__cpf_cnpj = cpf_cnpj
        self.__endereco = endereco  
        self.__telefone = telefone

    def apresentar_dados(self):
        print("Nome: " + self.nome_oficial)
        print("CPF/CNPJ: " + self.cpf_cnpj)
        print("Endereço: " + str(self.endereco))  # imprimir o endereço como uma string
        print("Telefone: " + self.telefone)

    def verificar_tipo(self):
        return "Pessoa"

    @property
    def nome_oficial(self):
        return self._nome_oficial

    @nome_oficial.setter
    def nome_oficial(self, nome_oficial):
        self._nome_oficial = nome_oficial

    @property
    def cpf_cnpj(self):
        return self.__cpf_cnpj

    @cpf_cnpj.setter
    def cpf_cnpj(self, cpf_cnpj):
        self.__cpf_cnpj = cpf_cnpj

    @property
    def endereco(self):
        return self.__endereco

    @endereco.setter
    def endereco(self, endereco):
        self.__endereco = endereco

    @property
    def telefone(self):
        return self.__telefone

    @telefone.setter
    def telefone(self, telefone):
        self.__telefone = telefone

File: OOP_final_project_JAVA_version/src/exercicio_incremental_parte2/Pessoa.java
package exercicio_incremental_parte2;

public class Pessoa {
    protected String nome_oficial;
    protected String cpf_cnpj;
    protected Endereco endereco;
    protected String telefone;

    public Pessoa(String nome_oficial, String cpf_cnpj, Endereco endereco, String telefone) {
        this.nome_oficial = nome_oficial;
        this.cpf_cnpj = cpf_cnpj;
        this.endereco = endereco;
        this.telefone = telefone;
    }
    
    public Pessoa(String cpf_cnpj, Endereco endereco, String telefone) {
        this.cpf_cnpj = cpf_cnpj;
        this.endereco = endereco;
        this.telefone = telefone;
    }
    
    public void apresentarDados() {
        System.out.println("Nome: " + getNome_oficial());
        System.out.println("CPF/CNPJ: " + getCpf_cnpj());
        System.out.println("Endereço: " + getEndereco());
        System.out.println("Telefone: " + getTelefone());
    }
    
    public String verificarTipo() {
        return "Pessoa";
    }
    

    protected String getNome_oficial() {
        return nome_oficial;
    }

    public void setNome_oficial(String nome_oficial) {
        this.nome_oficial = nome_oficial;
    }

    protected String getCpf_cnpj() {
        return cpf_cnpj;
    }

    public void setCpf_cnpj(String cpf_cnpj) {
        this.cpf_cnpj = cpf_cnpj;
    }

    public Endereco getEndereco() {
        return endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }
    
    
    
}
